### PR TITLE
0.6.4-alpha

### PR DESCRIPTION
**Pull request by Bistard**
## Summary by Sourcery

Introduce an inspector window for debugging and enhance the recent opened feature with pinning and clearing capabilities. Refactor the main window service for better lifecycle management and improve the disk storage system. Enhance the i18n system with a new Webpack plugin for localization optimization. Add new tests and an i18n integration test to ensure build correctness.

New Features:
- Introduce a new inspector window feature for debugging, allowing users to inspect configurations, status, context keys, commands, shortcuts, colors, and menus.
- Add support for recent opened files and directories, with the ability to pin items and clear the list.

Enhancements:
- Refactor the main window service to support asynchronous window operations and lifecycle binding with owner windows.
- Enhance the disk storage system by removing synchronous storage classes and improving asynchronous operations.
- Improve the i18n system by introducing a new Webpack plugin for transforming localization keys to indices, reducing bundle size.

Build:
- Add a new Webpack plugin for i18n key-to-index transformation, optimizing localization handling.

CI:
- Introduce an i18n integration test to validate localization files and ensure correct build processes.

Tests:
- Add tests for the new ReplaceType utility type, ensuring correct type transformations in various scenarios.
- Enhance tests for configuration change events, ensuring correct behavior of affect and match methods.

### CODE REVIEW

**Review by AAsteria — DISMISSED** [review]
Looks good, approved!

**Review by AAsteria — APPROVED** [review]
Approved!

**Comment by Bistard** [comment]
@sourcery-ai review

**Review by AAsteria — DISMISSED** [review]
AP, good to merge!

**Review by Bistard — COMMENTED** [review]
review all parts excludes `recentOpen` functionality

**Review by AAsteria — DISMISSED** [review]
Reviewed!

File: src/code/electron/app.ts [hunk 1]
```diff
@@ -163,16 +172,17 @@ export class ApplicationInstance extends Disposable implements IApplicationInsta
...
 
-        // life-cycle-service: READY
-        this.lifecycleService.setPhase(LifecyclePhase.Ready);
+        const rawRecentPath = this.statusService.get<string[]>(StatusKey.OpenRecent, []);
+        const recentPath = Array.isArray(rawRecentPath) ? rawRecentPath : [];
 
         // retrieve last saved opened window status
-        const uriToOpen: URI[] = [];
-        const shouldRestore = this.configurationService.get<boolean>(WorkbenchConfiguration.RestorePrevious);
+        const uriToOpen: URI[] = recentPath.map(path => URI.fromFile(path));
```
Comment (Bistard) [inline src/code/electron/app.ts:183]: 这里原本是`uriToOpen: URI[] = []`. 但这里你改成了`recentPath`. 这样首先有两个问题：
1. 这里的uriToOpen的格式比较特殊，不仅仅需要uri本身，而且还有一些特殊的标记符，比如表达一个directory格式为: `directory_path|directory`. 表达file的格式为: `file_path|file(|gotoLine)`. 其中`(|gotoLine)`代表optional，还能填个数字代表打开这个文件的时候还自动goto某一行。
2. 其次第二个问题是：你写成`uriToOpen: URI[] = recentPath`的话，岂不是每次启动软件时，会打开所有的directory和file。目前而言，就先设计成：只打开最近的一个吧。

Comment (Bistard) [reply src/code/electron/app.ts]: 我后面会push挺多commits，专门修这个问题，写了一个新的service叫做`RecentOpenService`。背后的动机就是：
1. recentOpened系列的功能被分散得定义在了软件的各个地方，非常不方便维护，所以为了single responsibility principle，集成了一个service。

**Review by AAsteria — DISMISSED** [review]
Reviewed and approved!

**Pull request merged**